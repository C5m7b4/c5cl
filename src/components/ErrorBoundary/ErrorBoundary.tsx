import React, { ErrorInfo } from 'react';

export interface ErrorBoundaryProps {
  children: React.ReactNode;
}

export interface ErrorBoundaryState {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<
  ErrorBoundaryProps,
  ErrorBoundaryState
> {
  public state: ErrorBoundaryState = {
    hasError: false,
  };

  public static getDerivedStateFromError(_: Error): ErrorBoundaryState {
    return {
      hasError: true,
    };
  }

  public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log('Uncaught Error: ', error, errorInfo);
  }

  public render() {
    if (this.state.hasError) {
      return <h1>Oops, something went wrong with your component</h1>;
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
