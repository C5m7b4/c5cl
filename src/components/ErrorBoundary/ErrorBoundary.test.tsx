import React from 'react';
import { render } from '@testing-library/react';

import ErrorBoundary from './ErrorBoundary';

const TestCmp = () => {
  return (
    <div>
      <h2>I am a component</h2>
    </div>
  );
};

const ErrorCmp = () => {
  throw new Error('this is an error');
};

describe('ErrorBoundary', () => {
  test('should render correctly', () => {
    const { container } = render(
      <ErrorBoundary>
        <TestCmp />
      </ErrorBoundary>
    );
    expect(container).toMatchSnapshot();
  });
  test('should render an error', () => {
    try {
      const { container } = render(
        <ErrorBoundary>
          <ErrorCmp />
        </ErrorBoundary>
      );
    } catch (error) {}
  });
});
