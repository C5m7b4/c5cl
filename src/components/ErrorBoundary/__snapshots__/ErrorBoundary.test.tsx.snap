// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render correctly 1`] = `
<div>
  <div>
    <h2>
      I am a component
    </h2>
  </div>
</div>
`;
